File: java/collatz-conjecture/src/main/java/CollatzCalculator.java
class CollatzCalculator {

    int computeStepCount(int start) {
        if (start <= 0) {
            throw new IllegalArgumentException("Only positive integers are allowed");
        }

        int number = start;
        int steps = 0;

        while (number != 1) {
            number = number % 2 == 0 ? number / 2 : number * 3 + 1;
            steps++;
        }

        return steps;
    }

}

File: kotlin/collatz-conjecture/src/main/kotlin/CollatzCalculator.kt
object CollatzCalculator {
    fun computeStepCount(start: Int): Int {
        require(start > 0)

        if (start == 1) {
            return 0
        }

        return (if (start % 2 == 0) computeStepCount(start / 2) else computeStepCount(start * 3 + 1)) + 1
    }
}
